src/bitmessageqt/__init__.py

```diff
@@ -1,63 +1,78 @@
-from debug import logger
+# pylint: disable=too-many-lines,broad-except,too-many-instance-attributes,global-statement,too-few-public-methods
+# pylint: disable=too-many-statements,too-many-branches,attribute-defined-outside-init,too-many-arguments,no-member
+# pylint: disable=unused-argument,no-self-use,too-many-locals,unused-variable,too-many-nested-blocks
+# pylint: disable=too-many-return-statements,protected-access,super-init-not-called,non-parent-init-called
+"""
+Initialise the QT interface
+"""
+
+from debug import logger  # pylint: disable=wrong-import-order
+
+import hashlib
+import locale
+import os
+import random
+import string
 import sys
+import textwrap
+import time
+from datetime import datetime, timedelta
+
 
 try:
     from PyQt4 import QtCore, QtGui
     from PyQt4.QtNetwork import QLocalSocket, QLocalServer
-except Exception as err:
-    logmsg = 'PyBitmessage requires PyQt unless you want to run it as a daemon and interact with it using the API. You can download it from http://www.riverbankcomputing.com/software/pyqt/download or by searching Google for \'PyQt Download\' (without quotes).'
+except ImportError:
+    logmsg = (
+        'PyBitmessage requires PyQt unless you want to run it as a daemon and interact with it using the API. You can'
+        ' download it from http://www.riverbankcomputing.com/software/pyqt/download or by searching Google for '
+        '\'PyQt Download\' (without quotes).'
+    )
     logger.critical(logmsg, exc_info=True)
     sys.exit()
 
-from tr import _translate
-from addresses import decodeAddress, addBMIfNotPresent
-import shared
-from bitmessageui import Ui_MainWindow
-from bmconfigparser import BMConfigParser
-import defaults
-from namecoin import namecoinConnection
-from messageview import MessageView
-from migrationwizard import Ui_MigrationWizard
-from foldertree import (
-    AccountMixin, Ui_FolderWidget, Ui_AddressWidget, Ui_SubscriptionWidget,
-    MessageList_AddressWidget, MessageList_SubjectWidget,
-    Ui_AddressBookWidgetItemLabel, Ui_AddressBookWidgetItemAddress)
-from settings import Ui_settingsDialog
-import settingsmixin
-import support
-import locale
-import time
-import os
-import hashlib
-from pyelliptic.openssl import OpenSSL
-import textwrap
-import debug
-import random
+
 from sqlite3 import register_adapter
-import string
-from datetime import datetime, timedelta
-from helper_ackPayload import genAckPayload
-from helper_sql import sqlQuery, sqlExecute, sqlExecuteChunked, sqlStoredProcedure
+
+import debug  # pylint: disable=ungrouped-imports
```

Comment:
Please place imports at the starting and in a order

Reply:
Sadly impossible without changing functionality. We need our `logger` during the import of a third party module.

In the next commit I have grouped the debugs, placed them after the built-in imports and now there is only one marked as violating, which is as much as can be done.